### PR TITLE
Extract disk name picking to a function

### PR DESCRIPTION
**Pull request by tomchop**
This CL extracts picking of which disk names to copy to a function, so that we can override it if needed.

### CODE REVIEW

File: dftimewolf/lib/collectors/gcloud.py [hunk 1]
```diff
@@ -137,43 +154,48 @@ def SetUp(self,
           image_project=image_project,
           image_family=image_family)
 
-      if disk_names:
-        for name in disk_names:
+    except AccessTokenRefreshError as err:
```
Comment (Onager) [inline dftimewolf/lib/collectors/gcloud.py]: "as exception" per style guide

File: dftimewolf/lib/collectors/gcloud.py [hunk 2]
```diff
@@ -137,43 +154,48 @@ def SetUp(self,
...
+      self.state.AddError('Something is wrong with your gcloud access token.')
+      self.state.AddError(err, critical=True)
+
+    except ApplicationDefaultCredentialsError as err:
+      self.state.AddError(
+          'Something is wrong with your Application Default Credentials. '
+          'Try running:\n  $ gcloud auth application-default login')
+      self.state.AddError(err, critical=True)
+
+  def FindDisksToCopy(self):
```
Comment (Onager) [inline dftimewolf/lib/collectors/gcloud.py]: Could this be private (_FindDisks...) and return a list of disks? It'd be a bit easier to follow the code, test etc.

File: dftimewolf/lib/collectors/gcloud.py [hunk 3]
```diff
@@ -137,43 +154,48 @@ def SetUp(self,
...
+    """Determines which disks to copy depending on object attributes."""
+    if not (self.remote_instance_name or self.disk_names):
+      self.state.AddError(
+          'You need to specify at least an instance name or disks to copy',
+          critical=True)
+      return
+
+    try:
```
Comment (Onager) [inline dftimewolf/lib/collectors/gcloud.py]: This is pretty heavily nested and hard to follow, can you please refactor this? Maybe it's worth breaking out some functions here - _GetDisksFromNames() maybe?

File: dftimewolf/lib/collectors/gcloud.py [hunk 4]
```diff
@@ -184,5 +206,8 @@ def SetUp(self,
             'disk name?')
```
Comment (Onager) [inline dftimewolf/lib/collectors/gcloud.py]: It would be helpful to have the name of the resource you're trying to retrieve here

Comment (tomchop) [reply dftimewolf/lib/collectors/gcloud.py]: The full detailed exception is now also added in the output.
